datadog/internal/validators/validators.go

```diff
@@ -303,3 +303,27 @@ func Float64Between(min, max float64) validator.String {
 		max: max,
 	}
 }
+
+func ValidateHeaders(val any, p cty.Path) diag.Diagnostics {
+	var diags diag.Diagnostics
+	value := val.(map[string]interface{})
+	for _, v := range value {
+		valueString, ok := v.(string)
+		if !ok {
+			diags = append(diags, diag.Diagnostic{
+				Severity:      diag.Error,
+				Summary:       "Invalid value",
+				Detail:        "Synthetics test header must be a string",
+				AttributePath: p,
+			})
+		} else if strings.Contains(valueString, "\n") || strings.Contains(valueString, "\r\n") {
```

Comment:
```suggestion
		} else if strings.Contains(valueString, "\n") {
```
I think the `\n` case is a superset of `\r\n`. Also, we should get a review from Synthetics confirming if there are any other invalid characters we should blacklist.

Reply:
http headers don't work with a blocklist, but with an allowlist. Basically, any visible character is allowed, everything else is forbidden: https://stackoverflow.com/a/75998796
But since this schema is not enforce on the backend, I guess it's fine to accept anything here as well 🤷